pylsp/workspace.py

```diff
@@ -441,3 +484,40 @@ def sys_path(self, environment_path=None, env_vars=None):
         environment = self.get_enviroment(environment_path=environment_path, env_vars=env_vars)
         path.extend(environment.get_sys_path())
         return path
+
+
+class Notebook:
+    """Represents a notebook."""
+    def __init__(self, uri, notebook_type, workspace, cells=None, version=None, metadata=None):
+        self.uri = uri
+        self.notebook_type = notebook_type
+        self.workspace = workspace
+        self.version = version
+        self.cells = cells or []
+        self.metadata = metadata or {}
+
+    def __str__(self):
+        return "Notebook with URI '%s'" % str(self.uri)
+
+    def add_cells(self, new_cells: List, start: int) -> None:
+        self.cells[start:start] = new_cells
+
+    def remove_cells(self, start: int, delete_count: int) -> None:
+        del self.cells[start:start+delete_count]
+
+
+# We inherit from Document for now to get the same API. However, cell document differ from the text documents in that
+# they have a language id.
+class Cell(Document):
+    """Represents a cell in a notebook."""
```

Comment:
```suggestion
class Cell(Document):
    """
    Represents a cell in a notebook.

    Notes
    -----
    We inherit from Document for now to get the same API. However, a cell document differ from
    text documents in that they have a language id.
    """
```

Move comment to be a note.

Reply:
Done